This macro breaks under autoconf 2.70. Since we don't worry abot darwin or
openbsd builds, simply remove it.

Upstream-Status: Pending
RP - 2021/1/28

Index: ruby-2.7.2/configure.ac
===================================================================
--- ruby-2.7.2.orig/configure.ac
+++ ruby-2.7.2/configure.ac
@@ -170,23 +170,6 @@ dnl     orig_cxxflags="$cxxflags"
 dnl     cxxflags="$cxxflags "'${optflags} ${debugflags} ${warnflags}'
 dnl ])
 
-AS_CASE(["$host_os:$build_os"],
-[darwin*:darwin*], [
-    AC_CHECK_TOOLS(CC, [clang gcc cc])
-    # Following Apple deployed clang are broken
-    # clang version 1.0 (http://llvm.org/svn/llvm-project/cfe/tags/Apple/clang-23 exported)
-    # Apple clang version 2.0 (tags/Apple/clang-137) (based on LLVM 2.9svn)
-    # Apple clang version 2.1 (tags/Apple/clang-163.7.1) (based on LLVM 3.0svn)
-    AS_IF([! $CC -E -xc - <<SRC >/dev/null], [
-	@%:@if defined __APPLE_CC__ && defined __clang_major__ && __clang_major__ < 3
-	@%:@error premature clang
-	@%:@endif
-SRC
-	AC_MSG_ERROR([clang version 3.0 or later is required])
-    ])],
-[openbsd*:openbsd*], [
-    AC_CHECK_TOOLS(CC, [cc])
-])
 AS_IF([test x"${build}" != x"${host}"], [
   AC_CHECK_TOOL(CC, gcc)
 ])
